packages/appeals-service-api/src/controllers/token.js

```diff
@@ -31,32 +44,147 @@ async function tokenPut(req, res) {
 	res.status(200).send({});
 }
 
-//checks code against id
+/**
+ * sends a code to user email
+ * @type {import('express').Handler}
+ */
+async function tokenPutV2(req, res) {
+	const { id, action } = req.body;
+	let { emailAddress } = req.body;
+
+	let user;
+
+	if (!emailAddress) {
+		// look up email from appeal
+		const savedAppeal = await getAppeal(id);
+		emailAddress = savedAppeal.email;
+
+		// check if user exists already
+		user = await appealUserRepository.getByEmail(emailAddress);
+
+		// create new user for email in appeal
+		if (!user) {
+			user = await appealUserRepository.createUser({
+				email: emailAddress,
+				isEnrolled: false
+			});
+		}
+	} else {
+		user = await appealUserRepository.getByEmail(emailAddress);
+	}
+
+	if (!user) {
+		throw ApiError.userNotFound();
+	}
+
+	const tokenCreatedAt = await getTokenCreatedAt(user.id);
+
+	if (tokenCreatedAt) {
+		// to avoid issue with multiple requests sending multiple emails
+		const secondsSinceTokenCreation = getMilliSecondsSinceDate(tokenCreatedAt);
+		if (secondsSinceTokenCreation < MILLISECONDS_BETWEEN_TOKENS) {
+			res.status(200).send({});
+			return;
+		}
+	}
+
+	const token = await createOrUpdateToken(user.id, action);
+	await sendSecurityCodeEmail(emailAddress, token, id);
+
+	res.status(200).send({});
+}
+
+/**
+ * @param {Date} date
+ * @returns {number}
+ */
+function getMilliSecondsSinceDate(date) {
+	return new Date().getTime() - new Date(date).getTime();
+}
+
+/**
+ * checks code against id
+ * @type {import('express').Handler}
+ */
 async function tokenPost(req, res) {
 	const { id, token } = req.body;
 
 	const document = await getTokenDocumentIfExists(id);
 
 	if (!document) {
-		return res.status(200).send({});
+		res.status(200).send({});
+		return;
```

Comment:
Curious, is there an advantage of doing the return on a different line or just to make sure we don't return any values?

Reply:
Since adding the jsdoc for express handler it gives me red squiggles:
Type 'Response<any, Record<string, any>, number>' is not assignable to type 'void'.

I don't think it actually matters though